core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1587,20 +1590,23 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     if (retentionSize < 0 || size < retentionSize) return 0
     var diff = size - retentionSize
     def shouldDelete(segment: LogSegment, nextSegmentOpt: Option[LogSegment]): Boolean = {
-      if (diff - segment.size >= 0) {
+      val shouldDelete = diff - segment.size >= 0
+      debug(s"$segment retentionSize breached: $shouldDelete, log size before delete segment=$diff, after delete segment=${diff - segment.size}")
+      if (shouldDelete) {
         diff -= segment.size
-        true
-      } else {
-        false
       }
+      shouldDelete
     }
 
     deleteOldSegments(shouldDelete, RetentionSizeBreach(this, remoteLogEnabled()))
   }
 
   private def deleteLogStartOffsetBreachedSegments(): Int = {
     def shouldDelete(segment: LogSegment, nextSegmentOpt: Option[LogSegment]): Boolean = {
-      nextSegmentOpt.exists(_.baseOffset <= (if (remoteLogEnabled()) localLogStartOffset() else logStartOffset))
+      val shouldDelete = nextSegmentOpt.exists(_.baseOffset <= (if (remoteLogEnabled()) localLogStartOffset() else logStartOffset))
+      debug(s"$segment logStartOffset breached: $shouldDelete, nextSegmentOpt=$nextSegmentOpt, " +
+        s"${if (remoteLogEnabled()) s"localLogStartOffset=${localLogStartOffset()}" else s"logStartOffset=$logStartOffset"}")
```

Comment:
ditto for `localLogStartOffset`

Reply:
Thanks for the review, I've addressed everything you mentioned.